### PR TITLE
Code cleanup for DPS clients

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: provisioning/service/src/Config/AttestationMechanism.cs [hunk 1]
```diff
@@ -129,111 +82,47 @@ private AttestationMechanism(AttestationMechanismType type, TpmAttestation tpm,
         /// <returns>The attestation of this object.</returns>
         public Attestation GetAttestation()
         {
-            switch (Type)
+            return Type switch
```
Comment (timtay-microsoft) [inline provisioning/service/src/Config/AttestationMechanism.cs:85]: This is pretty neat

**Review by timtay-microsoft — APPROVED** [review]

File: provisioning/service/src/Config/AttestationMechanism.cs (hunk 1 above)
Comment (drwill-ms) [reply provisioning/service/src/Config/AttestationMechanism.cs:85]: yeah, a newer C# feature. I'm not all the way used to it, but getting more used to seeing it.

**Review by brycewang-microsoft — APPROVED** [review]

**Pull request merged**